InnerEye/Azure/azure_util.py

```diff
@@ -108,28 +108,20 @@ def fetch_run(workspace: Workspace, run_recovery_id: str) -> Run:
     return run_to_recover
 
 
-def fetch_run_for_experiment(experiment_to_recover: Experiment, run_id_or_number: str) -> Run:
+def fetch_run_for_experiment(experiment_to_recover: Experiment, run_id: str) -> Run:
     """
     :param experiment_to_recover: an experiment
-    :param run_id_or_number: a string representing the Run ID or Run Number of one of the runs of the experiment
+    :param run_id: a string representing the Run ID of one of the runs of the experiment
     :return: the run matching run_id_or_number; raises an exception if not found
     """
     available_runs = experiment_to_recover.get_runs()
```

Comment:
it's possible that this is slow, can we move that inside of the exception handling? that's the only time we need it